src/domains/ecommerce-app/api/ecommerce-io/v2/_get_methods_redirect.xml.tpl

```diff
@@ -0,0 +1,134 @@
+          </set-header>
+          <set-header name="Content-Type" exists-action="override">
+              <value>application/json</value>
+          </set-header>
+          <set-body>@{
+            var userId = ((string)context.Variables.GetValueOrDefault("xUserId",""));
+            return new JObject(
+                    new JProperty("audience", "ecommerce"),
+                    new JProperty("duration", 900),
+                    new JProperty("privateClaims", new JObject(
+                        new JProperty("userId", userId)
+                    ))
+                ).ToString();
+          }</set-body>
+      </send-request>
+    <choose>
+        <when condition="@(((IResponse)context.Variables["x-jwt-token"]).StatusCode != 200)">
+            <return-response>
+                <set-status code="502" reason="Bad Gateway" />
+            </return-response>
+        </when>
+    </choose>
+    <set-variable name="token" value="@( (string) ( ((IResponse)context.Variables["x-jwt-token"] ).Body.As<JObject>(preserveContent: true)) ["token"])" />
+    <!-- Token JWT END-->
+    <set-variable name="baseUrl" value="@{
+        string ecommerceMethodsRedirectUrlMap = "{{ecommerce-methods-redirect-url-map}}";
+        string paymentTypeCode = context.Variables.GetValueOrDefault<string>("paymentTypeCode", string.Empty);
+        if (string.IsNullOrEmpty(ecommerceMethodsRedirectUrlMap) || string.IsNullOrEmpty(paymentTypeCode))
+        {
+            return null;
+        }
+        JObject urlsObject = JObject.Parse(ecommerceMethodsRedirectUrlMap);
+        return urlsObject.GetValue(paymentTypeCode, StringComparison.OrdinalIgnoreCase)?.ToString();
+    }" />
+    <choose>
+        <when condition="@(string.IsNullOrEmpty((string)context.Variables["baseUrl"]))">
+            <return-response>
+                <set-status code="404" reason="Not Found" />
+                <set-header name="Content-Type" exists-action="override">
+                    <value>application/json</value>
+                </set-header>
+                <set-body>@{
+                    return new JObject(
+                        new JProperty("error", "Configuration Not Found"),
+                        new JProperty("message", $"No redirect URL configured for the payment type: '{context.Variables.GetValueOrDefault<string>("paymentTypeCode", "N/A")}'.")
+                    ).ToString();
+                }</set-body>
+            </return-response>
+        </when>
+        <otherwise>
+            <return-response>
+              <set-header name="Content-Type" exists-action="override">
+                  <value>application/json</value>
+              </set-header>
+              <set-body>@{
+                  string sessionToken = context.Variables.GetValueOrDefault<string>("token", string.Empty);
+                  string finalRedirectUrl = (string)context.Variables["baseUrl"] + "&sessionToken=" + sessionToken;
```

Comment:
here we have to set session token as url fragment.

How we want to populate the configuration map? I expect something like `https://checkout.pagopa.it/ecommerce-fe/onboarding-payment-choice` (do not mind the ugly route name  😂)

So here we should concatenate session token to url doing something like #sessionToken=...
right?
```suggestion
                  string finalRedirectUrl = (string)context.Variables["baseUrl"] + "#sessionToken=" + sessionToken;
```